### Network Connections
Another starting point and deeper dive for associated processes are the systems network connections the following commands will cover how to enumerate connection data structures in memory
#### connections

In [10]:
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 connections

Volatility Foundation Volatility Framework 2.6
Offset(V)  Local Address             Remote Address            Pid
---------- ------------------------- ------------------------- ---
0x81e611f8 192.168.157.10:1053       218.85.133.23:89          796
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 connectio

Performaing a whois lookup on the IP `218.85.133.23`, from domain tools located here: http://whois.domaintools.com/218.85.133.23

```
IP Information for 218.85.133.23
 Quick Stats
IP Location	China China Fuzhou Chinanet Fujian Province Network
ASN	China AS133775 CHINATELECOM-FUJIAN-XIAMEN-IDC1 Xiamen, CN (registered Oct 28, 2014)
Whois Server	whois.apnic.net
IP Address	218.85.133.23
inetnum:        218.85.0.0 - 218.86.127.255
netname:        CHINANET-FJ
descr:          CHINANET Fujian province network
descr:          Data Communication Division
descr:          China Telecom
country:        CN
admin-c:        CH93-AP
tech-c:         CA67-AP
mnt-by:         MAINT-CHINANET
mnt-lower:      MAINT-CHINANET-FJ
status:         ALLOCATED NON-PORTABLE
last-modified:  2008-09-04T06:51:23Z
source:         APNIC

role:           CHINANETFJ IP ADMIN
address:        7,East Street,Fuzhou,Fujian,PRC
country:        CN
phone:          +86-591-83309761
fax-no:         +86-591-83371954
e-mail:         
remarks:        send spam reports  and abuse reports
remarks:        to 
remarks:        Please include detailed information and
remarks:        times in UTC
admin-c:        FH71-AP
tech-c:         FH71-AP
nic-hdl:        CA67-AP
remarks:        www.fjtelecom.com
notify:         
mnt-by:         MAINT-CHINANET-FJ
last-modified:  2011-12-06T00:10:50Z
source:         APNIC

person:         Chinanet Hostmaster
nic-hdl:        CH93-AP
e-mail:         
address:        No.31 ,jingrong street,beijing
address:        100032
phone:          +86-10-58501724
fax-no:         +86-10-58501724
country:        CN
mnt-by:         MAINT-CHINANET
last-modified:  2014-02-27T03:37:38Z
source:         APNIC
```

The connection was open to port 89 which is listed as SU MIT Telnet Gateway as the default service and because IE should be using 80 or 443 for HTTP and HTTPS connections. 

#### connscan
As with `pslist` and `psscan` volatility uses the `scan` naming convention to signify the plugin is performing a forensic search of the memory structure not just enumerating what the operating system would see. In this plugin the analysis is enumerating connection data structures from memory forensicly. 

In [17]:
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 connscan

Offset(P)  Local Address             Remote Address            Pid
---------- ------------------------- ------------------------- ---
0x0205ece0 192.168.157.10:1050       222.128.1.2:443           1672
0x020611f8 192.168.157.10:1053       218.85.133.23:89          796
0x032c01f8 192.168.157.10:1053       218.85.133.23:89          796
0x0337dce0 192.168.157.10:1050       222.128.1.2:443           1672
0x08a4ace0 192.168.157.10:1050       222.128.1.2:443           1672
0x18200ce0 192.168.157.10:1050       222.128.1.2:443           1672
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 connscan
Volatility Foundation Volatility Framework 2.6


This plugin exectuion finds another proess with an remote connection which was not previously found with the `connections` command. PID 1672 is explorer.exe which should not have any open connections to the Internet normally. Here is whois informaotin for the IP `222.128.1.2`:


```

IP Location	China China Beijing China Unicom Beijing Province Network
ASN	China AS4808 CHINA169-BJ China Unicom Beijing Province Network, CN (registered Jan 09, 1996)
Whois Server	whois.apnic.net
IP Address	222.128.1.2
inetnum:        222.128.0.0 - 222.131.255.255
netname:        UNICOM-BJ
descr:          China Unicom Beijing province network
descr:          China Unicom
country:        CN
admin-c:        CH1302-AP
tech-c:         SY21-AP
mnt-by:         APNIC-HM
mnt-lower:      MAINT-CNCGROUP-BJ
mnt-routes:     MAINT-CNCGROUP-RR
mnt-irt:        IRT-CU-CN
status:         ALLOCATED PORTABLE
last-modified:  2013-08-08T23:17:26Z
source:         APNIC

irt:            IRT-CU-CN
address:        No.21,Financial Street
address:        Beijing,100033
address:        P.R.China
e-mail:         
abuse-mailbox:  
admin-c:        CH1302-AP
tech-c:         CH1302-AP
auth:           # Filtered
mnt-by:         MAINT-CNCGROUP
last-modified:  2017-10-23T05:59:13Z
source:         APNIC

person:         ChinaUnicom Hostmaster
nic-hdl:        CH1302-AP
e-mail:         
address:        No.21,Jin-Rong Street
address:        Beijing,100033
address:        P.R.China
phone:          +86-10-66259764
fax-no:         +86-10-66259764
country:        CN
mnt-by:         MAINT-CNCGROUP
last-modified:  2017-08-17T06:13:16Z
source:         APNIC

person:         sun ying
address:        fu xing men nei da jie 97, Xicheng District
address:        Beijing 100800
country:        CN
phone:          +86-10-66030657
fax-no:         +86-10-66078815
e-mail:         
nic-hdl:        SY21-AP
mnt-by:         MAINT-CNCGROUP-BJ
last-modified:  2009-06-30T08:42:48Z
source:         APNIC

route:          222.128.0.0/14
descr:          China Unicom Beijing Province Network
country:        CN
origin:         AS4808
mnt-by:         MAINT-CNCGROUP-RR
last-modified:  2016-05-20T01:24:03Z
source:         APNIC
```

#### sockets
As with process enumeration there are several ways to enumerate network connections from memory. Below is another plugin, looks for and parses a singly-linked list of socket structures from the tcpip.sys module and only works on XP and 2003 systems.

In [20]:
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 sockets

Volatility Foundation Volatility Framework 2.6
Offset(V)       PID   Port  Proto Protocol        Address         Create Time
---------- -------- ------ ------ --------------- --------------- -----------
0x822ef510      796   1053      6 TCP             0.0.0.0         2009-05-05 19:29:06 UTC+0000
0x81fa9008     1212   1900     17 UDP             192.168.157.10  2009-04-16 16:10:20 UTC+0000
0x822d25b0        4      0     47 GRE             0.0.0.0         2009-05-05 19:28:35 UTC+0000
0x81e59748      716    500     17 UDP             0.0.0.0         2009-04-16 16:10:16 UTC+0000
0x822d7220        4    139      6 TCP             192.168.157.10  2009-04-16 16:10:00 UTC+0000
0x8206ce98      464   1032      6 TCP             127.0.0.1       2009-04-16 16:10:21 UTC+0000
0x82305008        4    445      6 TCP             0.0.0.0         2009-04-16 16:10:00 UTC+0000
0x81df5298      968    135      6 TCP             0.0.0.0         2009-04-16 16:10:07 UTC+0000
0x82305258        4    137     17 UDP

From this we can see PID 796 is listing on port 1053 at `2009-05-05 19:29:06 UTC+0000`. Giving you another data point for your analysis, especially for temporally based enumeration.  

In [23]:
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 sockscan

Volatility Foundation Volatility Framework 2.6
Offset(P)       PID   Port  Proto Protocol        Address         Create Time
---------- -------- ------ ------ --------------- --------------- -----------
0x01f354c8     1088    123     17 UDP             127.0.0.1       2009-04-16 16:10:20 UTC+0000
0x01ff5298      968    135      6 TCP             0.0.0.0         2009-04-16 16:10:07 UTC+0000
0x02059748      716    500     17 UDP             0.0.0.0         2009-04-16 16:10:16 UTC+0000
0x021a9008     1212   1900     17 UDP             192.168.157.10  2009-04-16 16:10:20 UTC+0000
0x021a9658     1088    123     17 UDP             192.168.157.10  2009-04-16 16:10:20 UTC+0000
0x0226ce98      464   1032      6 TCP             127.0.0.1       2009-04-16 16:10:21 UTC+0000
0x023514f8     1212   1900     17 UDP             127.0.0.1       2009-04-16 16:10:20 UTC+0000
0x023554c8     1088   1028     17 UDP             127.0.0.1       2009-04-16 16:10:20 UTC+0000
0x02365320      716      0    255 Res

In [26]:
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 netscan

Volatility Foundation Volatility Framework 2.6
ERROR   : volatility.debug    : This command does not support the profile WinXPSP2x86
python ../volatility/vol.py -f ../APT.img --profile=WinXPSP2x86 netscan


This command errors because this plugin only works for Windows Vista and above profiles byt the other provided commands should find what you need for network connections for Windows XP images. 